pkg/controller/windowsmachineconfig/windowsmachineconfig_controller.go

```diff
@@ -166,10 +200,43 @@ func (r *ReconcileWindowsMachineConfig) getCloudProvider(instance *wmcapi.Window
 // Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
 func (r *ReconcileWindowsMachineConfig) Reconcile(request reconcile.Request) (reconcile.Result, error) {
 	log.Info("reconciling", "namespace", request.Namespace, "name", request.Name)
+	mayBeWMCOobject := false
+	// Fetch the machine instance
+	machine := &mapiv1.Machine{}
+	provisionedPhase := "Provisioned"
+	err := r.client.Get(context.TODO(), request.NamespacedName, machine)
+	if err != nil {
+		mayBeWMCOobject = true
+		// TODO: Remove this when we switch to machine-api completely
+		//if k8sapierrors.IsNotFound(err) {
+		//	// Request object not found, could have been deleted after reconcile request.
+		//	// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+		//	// Return and don't requeue
+		//
+		//	return reconcile.Result{}, nil
+		//
+		//}
+		//// Error reading the object - requeue the request.
+		//return reconcile.Result{}, err
+	}
+	if mayBeWMCOobject {
+		goto WMCO
+	}
```

Comment:
You can easily avoid the `goto` by moving 226-236 here and doing a negation check i.e. 
```
if !wmcObject {
...
    return ...
}
```